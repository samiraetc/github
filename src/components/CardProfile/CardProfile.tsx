import Button from '../Button/Button';

export const COMPONENT_ID = 'card-profile';

type CardProfileProps = {
  name?: string | string[];
  imageUrl?: string;
  biography?: string | string[];
  repos?: number;
  hasButton?: boolean;
  className?: string;
  onClick?: () => unknown;
};
const CardProfile = ({
  name = '',
  imageUrl = 'https://avatars.githubusercontent.com/u/9120451?v=4',
  biography = '',
  repos,
  hasButton = false,
  className = '',
  onClick,
}: CardProfileProps) => {
  return (
    <div className={`${COMPONENT_ID}__container ${className}`} id={COMPONENT_ID}>
      <img src={imageUrl} alt="profile image" className={`${COMPONENT_ID}__image`} />
      <p className={`${COMPONENT_ID}__name`}>{name}</p>
      <p className={`${COMPONENT_ID}__biography`}>{biography}</p>
      {repos && <p className={`${COMPONENT_ID}__repositories`}>repositories: {repos}</p>}
      {hasButton && <Button name="REPOSITORIES" onClick={onClick} className={`${COMPONENT_ID}__button`} />}
    </div>
  );
};

export default CardProfile;
