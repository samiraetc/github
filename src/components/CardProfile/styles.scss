.card-profile {
  &__container {
    padding: $github-spacing-20;
    background-color: $github-white;
    border-radius: $github-border-radius-20;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    -webkit-box-shadow: 4px 9px 23px -8px rgba(0, 0, 0, 0.19);
    -moz-box-shadow: 4px 9px 23px -8px rgba(0, 0, 0, 0.19);
    box-shadow: 4px 9px 23px -8px rgba(0, 0, 0, 0.19);
    width: 50vh;
    border: 1px solid rgba(45, 164, 78, 0.5);
    gap: 15px;
  }

  &__image {
    border-radius: 50%;
    width: 50%;
  }

  &__name {
    padding: $github-spacing-12;
    font-size: $github-font-size-25;
    font-weight: 500;
  }

  &__biography {
    font-size: $github-font-size-16;
    font-weight: 300;
    text-align: center;
  }

  &__repositories {
    font-size: $github-font-size-16;
    font-weight: 300;
    text-align: center;
  }
}
