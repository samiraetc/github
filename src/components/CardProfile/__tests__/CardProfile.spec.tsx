import React from 'react';
import { configure, fireEvent, render } from '@testing-library/react';

import CardProfile, { COMPONENT_ID } from '../CardProfile';

configure({
  testIdAttribute: 'id',
});

describe('CardProfile', () => {
  test('should render Button with button', () => {
    const { asFragment, getByText, getByTestId } = render(
      <CardProfile name="Matt Biitencourt" biography="Hello, this is my Github profile" hasButton />,
    );

    expect(getByText('Matt Biitencourt'));
    expect(getByText('REPOSITORIES'));
    expect(getByTestId(COMPONENT_ID)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render Button with button and click', () => {
    const { asFragment, getByText } = render(
      <CardProfile name="Matt Biitencourt" biography="Hello, this is my Github profile" hasButton />,
    );

    fireEvent.click(getByText('REPOSITORIES'));
    expect(asFragment()).toMatchSnapshot();
  });
});
