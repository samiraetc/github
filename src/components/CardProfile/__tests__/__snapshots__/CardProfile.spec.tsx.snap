// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardProfile should render Button with button 1`] = `
<DocumentFragment>
  <div
    class="card-profile__container "
    id="card-profile"
  >
    <img
      alt="profile image"
      class="card-profile__image"
      src="https://avatars.githubusercontent.com/u/9120451?v=4"
    />
    <p
      class="card-profile__name"
    >
      Matt Biitencourt
    </p>
    <p
      class="card-profile__biography"
    >
      Hello, this is my Github profile
    </p>
    <div
      class="button card-profile__button"
      id="button"
    >
      <button
        class="button__action"
      >
        REPOSITORIES
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardProfile should render Button with button and click 1`] = `
<DocumentFragment>
  <div
    class="card-profile__container "
    id="card-profile"
  >
    <img
      alt="profile image"
      class="card-profile__image"
      src="https://avatars.githubusercontent.com/u/9120451?v=4"
    />
    <p
      class="card-profile__name"
    >
      Matt Biitencourt
    </p>
    <p
      class="card-profile__biography"
    >
      Hello, this is my Github profile
    </p>
    <div
      class="button card-profile__button"
      id="button"
    >
      <button
        class="button__action"
      >
        REPOSITORIES
      </button>
    </div>
  </div>
</DocumentFragment>
`;
